lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1928,14 +1943,21 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
 
   case OpPhi: {
     auto Phi = static_cast<SPIRVPhi *>(BV);
+    unsigned PhiPairsSize = Phi->getPairs().size() / 2;
     auto LPhi = dyn_cast<PHINode>(mapValue(
-        BV, PHINode::Create(transType(Phi->getType()),
-                            Phi->getPairs().size() / 2, Phi->getName(), BB)));
+        BV, PHINode::Create(transType(Phi->getType()), PhiPairsSize,
+                                      Phi->getName(), BB)));
     Phi->foreachPair([&](SPIRVValue *IncomingV, SPIRVBasicBlock *IncomingBB,
                          size_t Index) {
+      PHINodeArgsSet.insert(IncomingV);
       auto Translated = transValue(IncomingV, F, BB);
+      if (!Translated) {
+        PHINodeArgsMap.insert(std::make_pair(IncomingV,
+                                             std::make_tuple(LPhi, IncomingBB, Index)));
+      } else {
       LPhi->addIncoming(Translated,
                         dyn_cast<BasicBlock>(transValue(IncomingBB, F, BB)));
+      }
```

Comment:
I suggest that we remove almost everything from this foreach loop - this will allow to remove changes in `SPIRVReader.cpp:1160`